tools/src/test/java/org/apache/kafka/tools/consumer/group/DescribeConsumerGroupTest.java

```diff
@@ -141,699 +162,773 @@ public void testPrintVersion(String quorum) {
+        try (AutoCloseable protocolConsumerGroupExecutor = consumerGroupClosable(groupProtocol, PROTOCOL_GROUP, TOPIC, Collections.emptyMap());
+             // note the group to be queried is a different (non-existing) group
+             ConsumerGroupCommand.ConsumerGroupService service = consumerGroupService(new String[]{"--bootstrap-server", clusterInstance.bootstrapServers(), "--describe", "--group", group});
+        ) {
+            Entry<Optional<ConsumerGroupState>, Optional<Collection<MemberAssignmentState>>> res = service.collectGroupMembers(group, false);
+            assertTrue(res.getKey().map(s -> s.equals(ConsumerGroupState.DEAD)).orElse(false) && res.getValue().map(Collection::isEmpty).orElse(false),
+                    "Expected the state to be 'Dead', with no members in the group '" + group + "'.");
+
+            Entry<Optional<ConsumerGroupState>, Optional<Collection<MemberAssignmentState>>> res2 = service.collectGroupMembers(group, true);
+            assertTrue(res2.getKey().map(s -> s.equals(ConsumerGroupState.DEAD)).orElse(false) && res2.getValue().map(Collection::isEmpty).orElse(false),
+                    "Expected the state to be 'Dead', with no members in the group '" + group + "' (verbose option).");
+        }
     }
 
-    @ParameterizedTest(name = TEST_WITH_PARAMETERIZED_QUORUM_AND_GROUP_PROTOCOL_NAMES)
-    @MethodSource({"getTestQuorumAndGroupProtocolParametersAll"})
-    public void testDescribeStateOfNonExistingGroup(String quorum, String groupProtocol) throws Exception {
+    @ClusterTemplate("generator")
+    public void testDescribeStateOfNonExistingGroup() throws Exception {
         String group = "missing.group";
-        createOffsetsTopic(listenerName(), new Properties());
+        createTopic(TOPIC);
 
         // run one consumer in the group consuming from a single-partition topic
-        addConsumerGroupExecutor(1, groupProtocol);
-        // note the group to be queried is a different (non-existing) group
-        String[] cgcArgs = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--describe", "--group", group};
-        ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs);
-
-        GroupState state = service.collectGroupState(group);
-        assertTrue(Objects.equals(state.state, ConsumerGroupState.DEAD) && state.numMembers == 0 &&
-                state.coordinator != null && !brokers().filter(s -> s.config().brokerId() == state.coordinator.id()).isEmpty(),
-            "Expected the state to be 'Dead', with no members in the group '" + group + "'."
-        );
+        GroupProtocol groupProtocol = clusterInstance.config().serverProperties().get(GroupCoordinatorConfig.NEW_GROUP_COORDINATOR_ENABLE_CONFIG).trim().equals("true") ? GroupProtocol.CONSUMER : GroupProtocol.CLASSIC;
+        try (AutoCloseable protocolConsumerGroupExecutor = consumerGroupClosable(groupProtocol, PROTOCOL_GROUP, TOPIC, Collections.emptyMap());
```

Comment:
It looks to me that we could put the logic of determining `groupProtocol` in `consumerGroupClosable` method, so that we don't have to do the same thing in each test. Ex: Add a `consumerGroupClosable` method, and replace the `groupProtocol` with `clusterInstance.config().serverProperties()`. So we can do the groupProtocol determination inside the `consumerGroupClosable` method:

```
private AutoCloseable consumerGroupClosable(Map<String, String> serverProperties, GroupProtocol protocol, String groupId, String topicName) {
    GroupProtocol groupProtocol = serverProperties.get(GroupCoordinatorConfig.NEW_GROUP_COORDINATOR_ENABLE_CONFIG).trim().equals("true") ? GroupProtocol.CONSUMER : GroupProtocol.CLASSIC;
    return   consumerGroupClosable(groupProtocol, ...);
```

Reply:
Thanks for the suggestion. I found that I only run consumer group protocol when `group.coordinator.new.enable` is true. I change to use `ClusterInstance#supportedGroupProtocols` to check supported protocols, so we can run all cases like before. Thanks.